content/language/r/develop.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Use containers for R development
+keywords: R, local, development
+description: Learn how to develop your R application locally.
+---
+
+## Prerequisites
+
+Complete [Containerize a R application](containerize.md).
+
+## Overview
+
+In this section, you'll learn how to set up a development environment for your containerized application. This includes:
+
+- Adding a local database and persisting data
+- Configuring Compose to automatically update your running Compose services as you edit and save your code
+
+## Get the sample application
+
+You'll need to clone a new repository to get a sample application that includes logic to connect to the database.
+
+Change to a directory where you want to clone the repository and run the following command.
+
+```console
+$ git clone https://github.com/mfranzon/r-docker-dev.git
+```
+
+
+## Add a local database and persist data
+
+You can use containers to set up local services, like a database. In this section, you'll update the `compose.yaml` file to define a database service and a volume to persist data.
+
+In the cloned repository's directory, open the `compose.yaml` file in an IDE or text editor. 
+
+In the `compose.yaml` file, you need to uncomment all of the database instructions. In addition, you need to add the database password file as an environment variable to the server service and specify the secret file to use .
+
+The following is the updated `compose.yaml` file.
+
+```yaml {hl_lines="7-36"}
+services:
+  shiny-app:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - 3838:3838
+    environment:
+      - POSTGRES_PASSWORD_FILE=/run/secrets/db-password
+    depends_on:
+      db:
+        condition: service_healthy
+    secrets:
+      - db-password
+  db:
+    image: postgres
+    restart: always
+    user: postgres
+    secrets:
+      - db-password
+    volumes:
+      - db-data:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_DB=example
+      - POSTGRES_PASSWORD_FILE=/run/secrets/db-password
+    expose:
+      - 5432
+    healthcheck:
+      test: [ "CMD", "pg_isready" ]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+volumes:
+  db-data:
+secrets:
+  db-password:
+    file: db/password.txt
+```
+
+> **Note**
+>
+> To learn more about the instructions in the Compose file, see [Compose file
+> reference](/compose/compose-file/).
+
+Before you run the application using Compose, notice that this Compose file specifies a `password.txt` file to hold the database's password. You must create this file as it's not included in the source repository.
+
+In the cloned repository's directory, create a new directory named `db` and inside that directory create a file named `password.txt` that contains the password for the database. Using your favorite IDE or text editor, add the following contents to the `password.txt` file.
```

Comment:
This directory already existed so let's make sure we delete it from the template/starter repo